src/tests/utils/events.cairo

```diff
@@ -0,0 +1,90 @@
+use snforge_std::cheatcodes::events::EventFetcher;
+use snforge_std::{spy_events, SpyOn, Event, EventSpy, EventAssertions};
+use starknet::ContractAddress;
+
+pub fn spy_on(contract_address: ContractAddress) -> EventSpy {
+    spy_events(SpyOn::One(contract_address))
+}
+
+#[generate_trait]
+pub impl EventSpyExtImpl of EventSpyExt {
+    fn assert_only_event<T, +starknet::Event<T>, +Drop<T>>(
+        ref self: EventSpy, from_address: ContractAddress, event: T
+    ) {
+        self.assert_emitted_single(from_address, event);
+        self.assert_no_events_left_from(from_address);
+    }
+
+    fn assert_emitted_single<T, +starknet::Event<T>, +Drop<T>>(
+        ref self: EventSpy, from_address: ContractAddress, expected_event: T
+    ) {
+        self.assert_emitted(@array![(from_address, expected_event)]);
+    }
+
+    fn drop_event_from(ref self: EventSpy, from_address: ContractAddress) {
+        self.drop_events_from(from_address, 1);
+    }
+
+    fn drop_all_events(ref self: EventSpy) {
+        self.fetch_events();
+        self.events = array![];
+    }
+
+    fn drop_all_events_from(ref self: EventSpy, from_address: ContractAddress) {
+        self.fetch_events();
+        let mut new_events: Array<(ContractAddress, Event)> = array![];
+        while let Option::Some((from, event)) = self
+            .events
+            .pop_front() {
+                if from != from_address {
+                    new_events.append((from, event));
+                }
+            };
+        self.events = new_events;
+    }
+
+    fn drop_events_from(ref self: EventSpy, from_address: ContractAddress, number_to_drop: u32) {
+        self.fetch_events();
+        let mut dropped_number = 0;
+        let mut new_events: Array<(ContractAddress, Event)> = array![];
+        while let Option::Some((from, event)) = self
+            .events
+            .pop_front() {
+                if from == from_address && dropped_number < number_to_drop {
```

Comment:
I think it makes sense to define this right under `drop_event_from` to make this module (and eventually documentation) even tidier. Part of me also wonders if this should be renamed `drop_n_events_from` (and reverse the parameter order). WDYT?

Reply:
I like the `drop_n_events_from` name for the function, it makes the difference to the `drop_event_from` more explicit. I've also moved the function to place them next to each other

Regarding reversing the parameter order, I reckon it's better to leave it as it is. The `from_address` param is the dominant one here, and moreover that keeps parameter-order approach consistent within the module

Reply:
> Regarding reversing the parameter order, I reckon it's better to leave it as it is. The from_address param is the dominant one here, and moreover that keeps parameter-order approach consistent within the module

Fair points, sounds good